engine/src/main/java/org/terasology/rendering/opengl/GLSLMaterial.java

```diff
@@ -72,11 +73,13 @@
     private ShaderParameters shaderParameters;
 
     private DisposalAction disposalAction;
+    private MaterialData materialData;
 
     public GLSLMaterial(ResourceUrn urn, AssetType<?, MaterialData> assetType, MaterialData data) {
         super(urn, assetType);
         disposalAction = new DisposalAction(urn);
         getDisposalHook().setDisposeAction(disposalAction);
+        this.materialData = data;
         shaderManager = CoreRegistry.get(ShaderManager.class);
```

Comment:
Pinging @flo for advice on future cases like this: here is a CoreRegistry usage. In this case, what would be the best way to get the Context instance? Or should we use the injection decorator to fill the shaderManager field?